Client/Frontend/Home/TopSites/TopSitesViewModel.swift

```diff
@@ -228,15 +228,23 @@ extension TopSitesViewModel: HomepageSectionHandler {
                    at indexPath: IndexPath) -> UICollectionViewCell {
         if let cell = collectionView.dequeueReusableCell(cellType: TopSiteItemCell.self, for: indexPath),
            let contentItem = topSites[safe: indexPath.row] {
-            let favicon = topSitesDataAdaptor.getFaviconImage(forSite: contentItem.site)
+            let id = Int(arc4random())
+            cell.tag = id
+            siteImageHelper.fetchImageFor(site: contentItem.site,
+                                          imageType: .favicon,
+                                          shouldFallback: false) { image in
+                guard cell.tag == id else { return }
+                cell.imageView.image = image
+            }
+
             var textColor: UIColor?
 
             if wallpaperManager.featureAvailable {
                 textColor = wallpaperManager.currentWallpaper.textColor
             }
 
             cell.configure(contentItem,
-                           favicon: favicon,
+                           favicon: nil,
```

Comment:
nit, could remove the `favicon` parameter in the configure method since it's not used anymore